File: seed/csharp-model/trace/src/SeedTraceClient/V2/Problem/BasicTestCaseTemplate.cs
using System.Text.Json.Serialization
using SeedTraceClient.V2

namespace SeedTraceClient.V2

public class BasicTestCaseTemplate
{
    [JsonPropertyName("templateId")]
    public string TemplateId { get; init; }

    [JsonPropertyName("name")]
    public string Name { get; init; }

    [JsonPropertyName("description")]
    public TestCaseImplementationDescription Description { get; init; }

    [JsonPropertyName("expectedValueParameterId")]
    public string ExpectedValueParameterId { get; init; }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/problem/types/basic_test_case_template.rb
# frozen_string_literal: true

module Seed
  module V2
    module Problem
      module Types
        class BasicTestCaseTemplate < Internal::Types::Model
          field :template_id, -> { String }, optional: false, nullable: false, api_name: "templateId"
          field :name, -> { String }, optional: false, nullable: false
          field :description, -> {
            Seed::V2::Problem::Types::TestCaseImplementationDescription
          }, optional: false, nullable: false
          field :expected_value_parameter_id, -> {
            String
          }, optional: false, nullable: false, api_name: "expectedValueParameterId"
        end
      end
    end
  end
end
